--- svgalib-1.9.20/utils/Makefile.orig	2005-03-18 11:59:49.000000000 +0100
+++ svgalib-1.9.20/utils/Makefile	2005-03-21 23:43:50.573388232 +0100
@@ -59,7 +59,7 @@
 	$(INSTALLPROG) $(UTILPROGS) $(UTILINSTALLDIR)
 
 .depend:
-	gcc $(INCLUDES) -MM $(patsubst %.o,$(srcdir)/utils/%.c,$(OBJECTS)) >.depend
+	$(CC) $(INCLUDES) -MM $(patsubst %.o,$(srcdir)/utils/%.c,$(OBJECTS)) >.depend
 
 #
 # include a dependency file if one exists
