diff -ur svgalib-1.9.25.orig/utils/gtf/gtfcalc.c svgalib-1.9.25/utils/gtf/gtfcalc.c
--- svgalib-1.9.25.orig/utils/gtf/gtfcalc.c	2005-07-10 21:33:16.000000000 +0200
+++ svgalib-1.9.25/utils/gtf/gtfcalc.c	2009-07-23 20:23:02.731643196 +0200
@@ -64,7 +64,7 @@
 
 /*-------------------------- Implementation -------------------------------*/
 
-static double round(double v)
+double round(double v)
 {
 	return floor(v + 0.5);
 }
