--- svgalib-1.9.25/Makefile.cfg.orig	2006-07-14 13:38:58.000000000 +0200
+++ svgalib-1.9.25/Makefile.cfg	2006-07-21 09:38:54.084440000 +0200
@@ -44,7 +44,7 @@
 # Common prefix for installation directories.
 # NOTE: This directory must exist when you start the install.
 TOPDIR=
-prefix = $(TOPDIR)/usr/local
+prefix = $(TOPDIR)/usr
 exec_prefix = $(prefix)
 
 # Directory where the shared stubs and static library will be installed.
@@ -96,7 +96,7 @@
 LIBC_MEMCPY = y
 
 # Uncomment this if you want to compile and install the static libs.
-# INSTALLSTATICLIB = installstaticlib
+INSTALLSTATICLIB = installstaticlib
 
 # Comment this out if you don't want to install the shared libs.
 # If you do not install the shared nor the static libs, 'make static'
@@ -320,7 +320,7 @@
 
 # Utilites used.
 AR	  = ar
-INSTALL_PROGRAM	= install -sm755
+INSTALL_PROGRAM	= install -m755
 INSTALL_SCRIPT	= install -m755
 INSTALL_SHLIB	= install -m755
 INSTALL_DATA	= install -m644
