--- svgalib-1.9.19/kernel/svgalib_helper/kernel26compat.h.orig	2005-02-12 16:09:23.955025000 +0100
+++ svgalib-1.9.19/kernel/svgalib_helper/kernel26compat.h	2005-02-12 16:14:31.369291192 +0100
@@ -41,6 +41,10 @@
 
 # define pci_present() 1
 
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,10)
+#  define pci_find_class pci_get_class
+# endif
+
 # ifdef CONFIG_DEVFS_FS
 typedef void* devfs_handle_t;
 # endif
