--- svgalib-1.9.25/Makefile.orig	2006-07-14 13:42:59.000000000 +0200
+++ svgalib-1.9.25/Makefile	2006-07-21 09:49:09.304440000 +0200
@@ -77,98 +77,77 @@
 .PHONY: indent-gnu 
 
 installheaders:
-	@echo Installing header files in $(includedir).
-	@mkdir -p $(includedir)
-	@cp $(SRCDIR)/src/vga.h $(includedir)/vga.h
-	@chmod a+r $(includedir)/vga.h
-	@cp $(SRCDIR)/gl/vgagl.h $(includedir)/vgagl.h
-	@chmod a+r $(includedir)/vgagl.h
-	@cp $(SRCDIR)/src/mouse/vgamouse.h $(includedir)/vgamouse.h
-	@chmod a+r $(includedir)/vgamouse.h
-	@cp $(SRCDIR)/src/joystick/vgajoystick.h $(includedir)/vgajoystick.h
-	@chmod a+r $(includedir)/vgajoystick.h
-	@cp $(SRCDIR)/src/keyboard/vgakeyboard.h $(includedir)/vgakeyboard.h
-	@chmod a+r $(includedir)/vgakeyboard.h
+	@echo Installing header files in $(DESTDIR)$(includedir).
+	@mkdir -p $(DESTDIR)$(includedir)
+	@cp $(SRCDIR)/src/vga.h $(DESTDIR)$(includedir)/vga.h
+	@chmod a+r $(DESTDIR)$(includedir)/vga.h
+	@cp $(SRCDIR)/gl/vgagl.h $(DESTDIR)$(includedir)/vgagl.h
+	@chmod a+r $(DESTDIR)$(includedir)/vgagl.h
+	@cp $(SRCDIR)/src/mouse/vgamouse.h $(DESTDIR)$(includedir)/vgamouse.h
+	@chmod a+r $(DESTDIR)$(includedir)/vgamouse.h
+	@cp $(SRCDIR)/src/joystick/vgajoystick.h $(DESTDIR)$(includedir)/vgajoystick.h
+	@chmod a+r $(DESTDIR)$(includedir)/vgajoystick.h
+	@cp $(SRCDIR)/src/keyboard/vgakeyboard.h $(DESTDIR)$(includedir)/vgakeyboard.h
+	@chmod a+r $(DESTDIR)$(includedir)/vgakeyboard.h
 
 installsharedlib: $(SHAREDLIBS) $(SVGALIBSHAREDSTUBS)
 	@echo Installing shared library image as \
-		$(addprefix $(sharedlibdir)/,$(notdir $(SHAREDLIBS))).
-	@mkdir -p ${sharedlibdir};
+		$(addprefix $(DESTDIR)$(sharedlibdir)/,$(notdir $(SHAREDLIBS))).
+	@mkdir -p $(DESTDIR)${sharedlibdir};
 	@for foo in $(notdir $(SHAREDLIBS)); do \
-		$(INSTALL_SHLIB) sharedlib/$$foo $(sharedlibdir)/$$foo; \
-		(cd $(sharedlibdir); \
+		$(INSTALL_SHLIB) sharedlib/$$foo $(DESTDIR)$(sharedlibdir)/$$foo; \
+		(cd $(DESTDIR)$(sharedlibdir); \
 		 ln -sf $$foo `echo $$foo | sed 's/\.so\..*/.so/'` ); \
 	done
-	@./fixldsoconf
-	-ldconfig
 
 installstaticlib: static
-	@echo Installing static libraries in $(libdir).
-	@mkdir -p $(libdir)
-	@$(INSTALL_DATA) staticlib/libvga.a $(libdir)/libvga.a
-	@chmod a+r $(libdir)/libvga.a
-	@$(INSTALL_DATA) staticlib/libvgagl.a $(libdir)/libvgagl.a
-	@chmod a+r $(libdir)/libvgagl.a
+	@echo Installing static libraries in $(DESTDIR)$(libdir).
+	@mkdir -p $(DESTDIR)$(libdir)
+	@$(INSTALL_DATA) staticlib/libvga.a $(DESTDIR)$(libdir)/libvga.a
+	@chmod a+r $(DESTDIR)$(libdir)/libvga.a
+	@$(INSTALL_DATA) staticlib/libvgagl.a $(DESTDIR)$(libdir)/libvgagl.a
+	@chmod a+r $(DESTDIR)$(libdir)/libvgagl.a
 
 installutils: textutils $(LRMI)
-	@if [ ! -d $(bindir) ]; then \
-		echo No $(bindir) directory, creating it.; \
-		mkdir -p $(bindir); \
-	fi
-	@echo Installing textmode utilities in $(bindir):
+	@mkdir -p $(DESTDIR)$(bindir)
+	@echo Installing textmode utilities in $(DESTDIR)$(bindir):
 	@echo "restorefont:      Save/restore textmode font."
-	@$(INSTALL_PROGRAM) utils/restorefont $(bindir)
+	@$(INSTALL_PROGRAM) utils/restorefont $(DESTDIR)$(bindir)
 	@echo "restorepalette:   Set standard VGA palette."
-	@$(INSTALL_PROGRAM) utils/restorepalette $(bindir)
+	@$(INSTALL_PROGRAM) utils/restorepalette $(DESTDIR)$(bindir)
 	@echo "dumpreg:          Write ASCII dump of SVGA registers."
-	@$(INSTALL_PROGRAM) utils/dumpreg $(bindir)
+	@$(INSTALL_PROGRAM) utils/dumpreg $(DESTDIR)$(bindir)
 	@echo "restoretextmode:  Save/restore textmode registers."
-	@$(INSTALL_PROGRAM) utils/restoretextmode $(bindir)
+	@$(INSTALL_PROGRAM) utils/restoretextmode $(DESTDIR)$(bindir)
 	@echo "textmode:         Script that tries to restore textmode."
-	@$(INSTALL_SCRIPT) utils/textmode $(bindir)
+	@$(INSTALL_SCRIPT) utils/textmode $(DESTDIR)$(bindir)
 	@echo "savetextmode:     Script that saves textmode information used by 'textmode'."
-	@$(INSTALL_SCRIPT) utils/savetextmode $(bindir)
+	@$(INSTALL_SCRIPT) utils/savetextmode $(DESTDIR)$(bindir)
 ifeq ($(LRMI),lrmi)
 	@echo "mode3:       Restore textmode by setting VESA mode 3."
-	@$(INSTALL_PROGRAM) lrmi-0.6m/mode3 $(bindir)
+	@$(INSTALL_PROGRAM) lrmi-0.6m/mode3 $(DESTDIR)$(bindir)
 endif
-	@echo "Installing keymap utilities in $(bindir):"
+	@echo "Installing keymap utilities in $(DESTDIR)$(bindir):"
 	@echo "svgakeymap:       Perl script that generates scancode conversion maps."
-	@$(INSTALL_SCRIPT) utils/svgakeymap $(bindir)
+	@$(INSTALL_SCRIPT) utils/svgakeymap $(DESTDIR)$(bindir)
 
 installconfig:
-	mkdir -p ${datadir};
-	@if [ \( -f /usr/local/lib/libvga.config -a ! -f $(datadir)/libvga.config \) ]; then \
-		echo "Moving old config file /usr/local/lib/libvga.config to $(datadir)." ; \
-		mv -f /usr/local/lib/libvga.config $(datadir)/libvga.config; \
-	fi
-	@if [ \( -f /usr/local/lib/libvga.et4000 -a ! -f $(datadir)/libvga.et4000 \) ]; then \
-		echo "Moving old config file /usr/local/lib/libvga.et4000 to $(datadir)." ; \
-		mv -f /usr/local/lib/libvga.et4000 $(datadir)/libvga.et4000; \
-	fi
-	@if [ \( -f /usr/local/lib/libvga.ega -a ! -f $(datadir)/libvga.ega \) ]; then \
-		echo "Moving old config file /usr/local/lib/libvga.ega to $(datadir)." ; \
-		mv -f /usr/local/lib/libvga.ega $(datadir)/libvga.ega; \
-	fi
-	@if [ \( -f /etc/mach32.eeprom -a ! -f $(datadir)/mach32.eeprom \) ]; then \
-		echo Consider moving your /etc/mach32.eeprom file to $(datadir) ; \
-		echo and changing $(datadir)/libvga.config appropriately. ; \
-	fi
-	@if [ ! -f $(datadir)/libvga.config ]; then \
-		echo Installing default configuration file in $(datadir).; \
-		cp $(CONFDIR)/libvga.config $(datadir)/libvga.config; \
+	mkdir -p $(DESTDIR)${datadir};
+	@if [ ! -f $(DESTDIR)$(datadir)/libvga.config ]; then \
+		echo Installing default configuration file in $(DESTDIR)$(datadir).; \
+		cp $(CONFDIR)/libvga.config $(DESTDIR)$(datadir)/libvga.config; \
 	fi
-	@if [ ! -f $(datadir)/libvga.et4000 ]; then \
-		echo Installing dynamically loaded ET4000 registers in $(datadir).; \
-		cp $(CONFDIR)/et4000.regs $(datadir)/libvga.et4000; \
+	@if [ ! -f $(DESTDIR)$(datadir)/libvga.et4000 ]; then \
+		echo Installing dynamically loaded ET4000 registers in $(DESTDIR)$(datadir).; \
+		cp $(CONFDIR)/et4000.regs $(DESTDIR)$(datadir)/libvga.et4000; \
 	fi
-	@if [ ! -f $(datadir)/default.keymap ]; then \
-		echo Installing default keymap file in $(datadir).; \
-		cp $(CONFDIR)/default.keymap $(datadir)/null.keymap; \
+	@if [ ! -f $(DESTDIR)$(datadir)/default.keymap ]; then \
+		echo Installing default keymap file in $(DESTDIR)$(datadir).; \
+		cp $(CONFDIR)/default.keymap $(DESTDIR)$(datadir)/null.keymap; \
 	fi
-	@if [ ! -f $(datadir)/dvorak-us.keymap ]; then \
-		echo Installing Dvorak keymap file in $(datadir).; \
-		cp $(CONFDIR)/dvorak-us.keymap $(datadir)/dvorak-us.keymap; \
+	@if [ ! -f $(DESTDIR)$(datadir)/dvorak-us.keymap ]; then \
+		echo Installing Dvorak keymap file in $(DESTDIR)$(datadir).; \
+		cp $(CONFDIR)/dvorak-us.keymap $(DESTDIR)$(datadir)/dvorak-us.keymap; \
 	fi
 
 installman:
--- svgalib-1.9.23/doc/Makefile.orig	2005-07-02 18:21:57.000000000 +0200
+++ svgalib-1.9.23/doc/Makefile	2005-10-10 13:38:10.000000000 +0200
@@ -26,7 +26,7 @@
 SVGALIB=$(subst man,$(srcdir)/doc/man,$(SVGALIB1))
 VGAGL=$(subst man,$(srcdir)/doc/man,$(VGAGL1))
 THREED=$(subst man,$(srcdir)/doc/man,$(THREED1))
-COMPMANPAGES=$(shell cd $(srcdir)/doc; find man? -name "*.gz" -type f -print)
+COMPMANPAGES=$(shell cd $(srcdir)/doc; find man? -name "*.[1-8]" -type f -print)
 
 .PHONY: clean ascii dvi ps install gunzip gzip
 
@@ -63,18 +63,12 @@
 	cat $(VGAGL) | grep -v '^\.so ' | troff -man -T$@ > vgagl.$@
 	cat $(THREED) | grep -v '^\.so ' | troff -man -T$@ > 3d.$@
 
-install: gzip
-	mkdir -p $(mandir)/man1 $(mandir)/man3 $(mandir)/man5 \
-		 $(mandir)/man6 $(mandir)/man7 $(mandir)/man8
+install:
+	install -d $(DESTDIR)$(mandir)/man1 $(DESTDIR)$(mandir)/man3 $(DESTDIR)$(mandir)/man5 \
+		 $(DESTDIR)$(mandir)/man6 $(DESTDIR)$(mandir)/man7 $(DESTDIR)$(mandir)/man8
 	for i in $(COMPMANPAGES); do\
-		$(INSTALL_DATA) $(srcdir)/doc/$$i $(mandir)/$$i; \
+		$(INSTALL_DATA) $(srcdir)/doc/$$i $(DESTDIR)$(mandir)/$$i; \
 	done
-	make gunzip
-ifdef MAKEWHATIS
-	@echo "Rebuilding the whatis database (this is slow!)"
-	@echo "You can comment this out in Makefile.cfg!"
-	$(MAKEWHATIS)
-endif
 
 uninstall:
 	@echo Removing all svgalib man pages.
diff -Nur svgalib-1.9.14.orig/lrmi-0.6m/Makefile svgalib-1.9.14/lrmi-0.6m/Makefile
--- svgalib-1.9.14.orig/lrmi-0.6m/Makefile	Sun Apr 16 16:05:09 2000
+++ svgalib-1.9.14/lrmi-0.6m/Makefile	Mon Apr 29 20:16:56 2002
@@ -25,8 +25,8 @@
 	$(CC) $(CFLAGS) -o $@ $^
 
 install: mode3 vga_reset
-	install mode3 /sbin
-	install vga_reset /sbin
+	install mode3 $(DESTDIR)/sbin
+	install vga_reset $(DESTDIR)/sbin
 
 .PHONY: clean
 clean:
--- svgalib-1.9.25/threeDKit/Makefile.orig	2006-07-14 13:30:49.000000000 +0200
+++ svgalib-1.9.25/threeDKit/Makefile	2006-07-21 09:51:39.854440000 +0200
@@ -55,18 +55,17 @@
 	$(CC) -shared -Wl,-soname,lib3dkit.so.$(MAJOR_VER) \
 	  -o lib3dkit.so.$(VERSION) $(OBJECTS) -lm
 
-$(sharedlibdir)/lib3dkit.so.$(VERSION): lib3dkit.so.$(VERSION)
-	$(INSTALL_SHLIB) $< $(sharedlibdir)/$<
-	(cd $(sharedlibdir); ln -sf lib3dkit.so.$(VERSION) `echo lib3dkit.so.$(VERSION) | sed 's/\.so\..*/.so/'` )
-	-ldconfig
+$(DESTDIR)$(sharedlibdir)/lib3dkit.so.$(VERSION): lib3dkit.so.$(VERSION)
+	$(INSTALL_SHLIB) $< $(DESTDIR)$(sharedlibdir)/$<
+	(cd $(DESTDIR)$(sharedlibdir); ln -sf lib3dkit.so.$(VERSION) `echo lib3dkit.so.$(VERSION) | sed 's/\.so\..*/.so/'` )
 
-install: $(sharedlibdir)/lib3dkit.so.$(VERSION) installheaders
+install: $(DESTDIR)$(sharedlibdir)/lib3dkit.so.$(VERSION) installheaders
 
 installheaders:
-	@cp 3dkit.h $(includedir)/3dkit.h
-	@chmod a+r $(includedir)/3dkit.h
-	@cp triangle.h $(includedir)/triangle.h
-	@chmod a+r $(includedir)/triangle.h
+	@cp 3dkit.h $(DESTDIR)$(includedir)/3dkit.h
+	@chmod a+r $(DESTDIR)$(includedir)/3dkit.h
+	@cp triangle.h $(DESTDIR)$(includedir)/triangle.h
+	@chmod a+r $(DESTDIR)$(includedir)/triangle.h
 
 endif	  
 
--- svgalib-1.9.23/kernel/svgalib_helper/Makefile.orig	2005-08-18 15:08:30.000000000 +0200
+++ svgalib-1.9.23/kernel/svgalib_helper/Makefile	2005-10-10 13:39:11.000000000 +0200
@@ -56,10 +56,9 @@
 	mknod -m 666 /dev/svga3 c $(SVGALIB_HELPER_MAJOR) 3
 	mknod -m 666 /dev/svga4 c $(SVGALIB_HELPER_MAJOR) 4
 
-install: device modules_install
+install: modules_install
 
 modules_install: $(MODNAME)
-	mkdir -p /lib/modules/$(VER)/kernel/misc
-	install -m 0644 -c $(TARGET).ko /lib/modules/$(VER)/kernel/misc ||	install -m 0644 -c $(TARGET).o /lib/modules/$(VER)/kernel/misc
-	depmod -a $(VER)
+	install -m 0755 -d $(DESTDIR)/lib/modules/$(VER)/kernel/misc
+	install -m 0644 -c $(TARGET).ko $(DESTDIR)/lib/modules/$(VER)/kernel/misc ||	install -m 0644 -c $(TARGET).o $(DESTDIR)/lib/modules/$(VER)/kernel/misc
 
--- svgalib-1.9.18/kernel/svgalib_helper/Makefile.alt.orig	2003-08-03 16:42:23.000000000 +0200
+++ svgalib-1.9.18/kernel/svgalib_helper/Makefile.alt	2003-10-07 18:21:46.267574328 +0200
@@ -58,11 +58,11 @@
 $(TARGET).ko: $(TARGET).o
 	$(LD) -d -r $^ -o $@
 
-install: device modules_install
+install: modules_install
 
 modules_install: $(OBJS)
-	mkdir -p /lib/modules/$(VER)/kernel/misc 
-	install -m 0644 -c $(OBJS) /lib/modules/$(VER)/kernel/misc
+	install -m 0755 -d $(DESTDIR)/lib/modules/$(VER)/kernel/misc 
+	install -m 0644 -c $(OBJS) $(DESTDIR)/lib/modules/$(VER)/kernel/misc
 
 device:
 	rm -f /dev/svgalib_helper* /dev/svga_helper* /dev/svga /dev/svga?
