--- svgalib-1.9.19/kernel/svgalib_helper/main.c.orig	Fri May  7 11:10:22 2004
+++ svgalib-1.9.19/kernel/svgalib_helper/main.c	Tue May 18 13:26:33 2004
@@ -694,5 +694,11 @@
 MODULE_LICENSE("GPL");
 #endif
 
+#if defined(MODULE_ALIAS_CHARDEV_MAJOR)
+MODULE_ALIAS_CHARDEV_MAJOR(SVGALIB_HELPER_MAJOR);
+#elif defined(MODULE_ALIAS)
+MODULE_ALIAS("char-major-" __stringify(SVGALIB_HELPER_MAJOR));
+#endif
+
 MODULE_AUTHOR("Matan Ziv-Av <matan@svgalib.org>");
 MODULE_DESCRIPTION("Generic hardware access to vga cards");
