--- svgalib-1.9.25/kernel/svgalib_helper/kernel26compat.h
+++ svgalib-1.9.25/kernel/svgalib_helper/kernel26compat.h
@@ -105,7 +109,8 @@
      class_device_create(svgalib_helper_class,                      	\
                              MKDEV(SVGALIB_HELPER_MAJOR, _minor),       \
                              &sh_pci_devs[_minor]->dev->dev, _name);
-#else /* 2.6.15 changed class_device_create */
+/* 2.6.15 changed class_device_create */
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(2,6,26)
 #  define SLH_SYSFS_ADD_CONTROL                                         \
      class_device_create(svgalib_helper_class, NULL,                	\
                              MKDEV(SVGALIB_HELPER_MAJOR, 0),            \
@@ -115,7 +120,18 @@
      class_device_create(svgalib_helper_class, NULL,                	\
                              MKDEV(SVGALIB_HELPER_MAJOR, _minor),       \
                              &sh_pci_devs[_minor]->dev->dev, _name);
-#endif /* 2.6.15 */
+/* 2.6.26 changed class_device_create to device_create */
+#else
+#  define SLH_SYSFS_ADD_CONTROL                                         \
+     device_create(svgalib_helper_class, NULL,                		\
+                             MKDEV(SVGALIB_HELPER_MAJOR, 0),            \
+                             "svga");
+
+#  define SLH_SYSFS_ADD_DEVICE(_name, _minor)                           \
+     device_create(svgalib_helper_class, &sh_pci_devs[_minor]->dev->dev, \
+                             MKDEV(SVGALIB_HELPER_MAJOR, _minor),       \
+                             _name);					
+#endif
 
 #  define SLH_SYSFS_REMOVE_DEVICE(i)                                    \
      class_destroy(svgalib_helper_class);
diff -ur svgalib-1.9.25.orig/src/lrmi.6.c svgalib-1.9.25/src/lrmi.6.c
--- svgalib-1.9.25.orig/src/lrmi.6.c	2005-10-22 13:29:17.000000000 +0200
+++ svgalib-1.9.25/src/lrmi.6.c	2008-08-05 10:33:54.000000000 +0200
@@ -169,6 +169,13 @@
 		}
 	}
 
+#ifndef TF_MASK
+#define TF_MASK X86_EFLAGS_TF
+#define IF_MASK X86_EFLAGS_IF
+#define IOPL_MASK X86_EFLAGS_IOPL
+#define VIF_MASK X86_EFLAGS_VIF
+#define VIP_MASK X86_EFLAGS_VIP
+#endif
 
 #define DEFAULT_VM86_FLAGS 	(IF_MASK | IOPL_MASK)
 #define DEFAULT_STACK_SIZE 	0x1000
diff -ur svgalib-1.9.25.orig/src/lrmi.9.c svgalib-1.9.25/src/lrmi.9.c
--- svgalib-1.9.25.orig/src/lrmi.9.c	2005-10-31 19:42:46.000000000 +0100
+++ svgalib-1.9.25/src/lrmi.9.c	2008-08-05 10:34:29.000000000 +0200
@@ -206,6 +206,13 @@
 
 
 #if defined(__linux__)
+#ifndef TF_MASK
+#define TF_MASK X86_EFLAGS_TF
+#define IF_MASK X86_EFLAGS_IF
+#define IOPL_MASK X86_EFLAGS_IOPL
+#define VIF_MASK X86_EFLAGS_VIF
+#define VIP_MASK X86_EFLAGS_VIP
+#endif
 #define DEFAULT_VM86_FLAGS 	(IF_MASK | IOPL_MASK)
 #elif defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
 #define DEFAULT_VM86_FLAGS  (PSL_I | PSL_IOPL)
diff -ur svgalib-1.9.25.orig/lrmi-0.6m/lrmi.c svgalib-1.9.25/lrmi-0.6m/lrmi.c
--- svgalib-1.9.25.orig/lrmi-0.6m/lrmi.c	2005-10-22 13:29:17.000000000 +0200
+++ svgalib-1.9.25/lrmi-0.6m/lrmi.c	2008-08-05 10:33:54.000000000 +0200
@@ -169,6 +169,13 @@
 		}
 	}
 
+#ifndef TF_MASK
+#define TF_MASK X86_EFLAGS_TF
+#define IF_MASK X86_EFLAGS_IF
+#define IOPL_MASK X86_EFLAGS_IOPL
+#define VIF_MASK X86_EFLAGS_VIF
+#define VIP_MASK X86_EFLAGS_VIP
+#endif
 
 #define DEFAULT_VM86_FLAGS 	(IF_MASK | IOPL_MASK)
 #define DEFAULT_STACK_SIZE 	0x1000
diff -ur svgalib-1.9.25.orig/lrmi-0.9/lrmi.c svgalib-1.9.25/lrmi-0.9/lrmi.c
--- svgalib-1.9.25.orig/lrmi-0.9/lrmi.c	2005-10-31 19:42:46.000000000 +0100
+++ svgalib-1.9.25/lrmi-0.9/lrmi.c	2008-08-05 10:34:29.000000000 +0200
@@ -206,6 +206,13 @@
 
 
 #if defined(__linux__)
+#ifndef TF_MASK
+#define TF_MASK X86_EFLAGS_TF
+#define IF_MASK X86_EFLAGS_IF
+#define IOPL_MASK X86_EFLAGS_IOPL
+#define VIF_MASK X86_EFLAGS_VIF
+#define VIP_MASK X86_EFLAGS_VIP
+#endif
 #define DEFAULT_VM86_FLAGS 	(IF_MASK | IOPL_MASK)
 #elif defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
 #define DEFAULT_VM86_FLAGS  (PSL_I | PSL_IOPL)
