--- svgalib-1.9.18/Makefile.orig	2002-12-22 11:58:47.000000000 +0100
+++ svgalib-1.9.18/Makefile	2003-10-07 18:17:06.368125512 +0200
@@ -77,125 +77,70 @@
 .PHONY: indent-gnu 
 
 installheaders:
-	@echo Installing header files in $(includedir).
-	@if [ -f /usr/include/vga.h ]; then \
-		rm -f /usr/include/vga.h /usr/include/vgagl.h; \
-		echo Old header files in /usr/include removed.; \
-	fi
-	@mkdir -p $(includedir)
-	@cp $(srcdir)/src/vga.h $(includedir)/vga.h
-	@chmod a+r $(includedir)/vga.h
-	@cp $(srcdir)/gl/vgagl.h $(includedir)/vgagl.h
-	@chmod a+r $(includedir)/vgagl.h
-	@cp $(srcdir)/src/mouse/vgamouse.h $(includedir)/vgamouse.h
-	@chmod a+r $(includedir)/vgamouse.h
-	@cp $(srcdir)/src/joystick/vgajoystick.h $(includedir)/vgajoystick.h
-	@chmod a+r $(includedir)/vgajoystick.h
-	@cp $(srcdir)/src/keyboard/vgakeyboard.h $(includedir)/vgakeyboard.h
-	@chmod a+r $(includedir)/vgakeyboard.h
+	@echo Installing header files in $(DESTDIR)$(includedir).
+	@install -d -m755 $(DESTDIR)$(includedir)
+	@install -m644 $(srcdir)/src/vga.h $(DESTDIR)$(includedir)/vga.h
+	@install -m644 $(srcdir)/gl/vgagl.h $(DESTDIR)$(includedir)/vgagl.h
+	@install -m644 $(srcdir)/src/mouse/vgamouse.h $(DESTDIR)$(includedir)/vgamouse.h
+	@install -m644 $(srcdir)/src/joystick/vgajoystick.h $(DESTDIR)$(includedir)/vgajoystick.h
+	@install -m644 $(srcdir)/src/keyboard/vgakeyboard.h $(DESTDIR)$(includedir)/vgakeyboard.h
 
 installsharedlib: $(SHAREDLIBS) $(SVGALIBSHAREDSTUBS)
-	@if [ -f /usr/lib/libvga.sa ]; then \
-		rm -f /usr/lib/libvga.sa /usr/lib/libvgagl.sa; \
-		echo Old stubs in /usr/lib removed.; \
-	fi
-ifndef KEEPSHAREDLIBS
-	@echo "Removing shared library images (old & current)..."
-	@for i in $(OBSOLETELDIRS); do \
-	    rm -f `echo $(OBSOLETESHAREDIMAGES) | sed s?/lib/?$$i?g`; \
-         done
-endif
 	@echo Installing shared library image as \
-		$(addprefix $(sharedlibdir)/,$(notdir $(SHAREDLIBS))).
+		$(addprefix $(DESTDIR)$(sharedlibdir)/,$(notdir $(SHAREDLIBS))).
+	@install -d -m755 $(DESTDIR)$(sharedlibdir)
 	@for foo in $(notdir $(SHAREDLIBS)); do \
-		$(INSTALL_SHLIB) sharedlib/$$foo $(sharedlibdir)/$$foo; \
-		(cd $(sharedlibdir); \
+		$(INSTALL_SHLIB) sharedlib/$$foo $(DESTDIR)$(sharedlibdir)/$$foo; \
+		(cd $(DESTDIR)$(sharedlibdir); \
 		 ln -sf $$foo `echo $$foo | sed 's/\.so\..*/.so/'` ); \
 	done
-	@./fixldsoconf
-	@ldconfig
 
 installstaticlib: static
-	@echo Installing static libraries in $(libdir).
-	@mkdir -p $(libdir)
-	@rm -f /usr/lib/libvga.a
-	@$(INSTALL_DATA) staticlib/libvga.a $(libdir)/libvga.a
-	@chmod a+r $(libdir)/libvga.a
-	@rm -f /usr/lib/libvgagl.a
-	@$(INSTALL_DATA) staticlib/libvgagl.a $(libdir)/libvgagl.a
-	@chmod a+r $(libdir)/libvgagl.a
+	@echo Installing static libraries in $(DESTDIR)$(libdir).
+	@install -d -m755 $(DESTDIR)$(libdir)
+	@$(INSTALL_DATA) staticlib/libvga.a $(DESTDIR)$(libdir)/libvga.a
+	@$(INSTALL_DATA) staticlib/libvgagl.a $(DESTDIR)$(libdir)/libvgagl.a
 
 installutils: textutils $(LRMI)
-	@if [ ! -d $(bindir) ]; then \
-		echo No $(bindir) directory, creating it.; \
-		mkdir -p $(bindir); \
-	fi
-	@if [ -f /usr/bin/restorefont ]; then \
-		echo Removing old utilities in /usr/bin.; \
-		for x in $(UTILS); do rm -f /usr/bin/$$x; done; \
-	fi
-	@if [ -f /usr/bin/convfont ]; then \
-		echo Removing inappropriate utilities in /usr/bin.; \
-		rm -f /usr/bin/convfont /usr/bin/setmclk; \
-	fi
-	@echo Installing textmode utilities in $(bindir):
+	@install -d $(DESTDIR)$(bindir)
+	@echo Installing textmode utilities in $(DESTDIR)$(bindir):
 	@echo "restorefont:      Save/restore textmode font."
-	@cp utils/restorefont $(bindir)
+	@cp utils/restorefont $(DESTDIR)$(bindir)
 	@echo "restorepalette:   Set standard VGA palette."
-	@cp utils/restorepalette $(bindir)
+	@cp utils/restorepalette $(DESTDIR)$(bindir)
 	@echo "dumpreg:          Write ASCII dump of SVGA registers."
-	@cp utils/dumpreg $(bindir)
+	@cp utils/dumpreg $(DESTDIR)$(bindir)
 	@echo "restoretextmode:  Save/restore textmode registers."
-	@cp utils/restoretextmode $(bindir)
+	@cp utils/restoretextmode $(DESTDIR)$(bindir)
 	@echo "textmode:         Script that tries to restore textmode."
-	@cp utils/textmode $(bindir)
+	@cp utils/textmode $(DESTDIR)$(bindir)
 	@echo "savetextmode:     Script that saves textmode information used by 'textmode'."
-	@cp utils/savetextmode $(bindir)
+	@cp utils/savetextmode $(DESTDIR)$(bindir)
 ifeq ($(LRMI),lrmi)
 	@echo "mode3:       Restore textmode by setting VESA mode 3."
-	@cp lrmi-0.6m/mode3 $(bindir)
+	@cp lrmi-0.6m/mode3 $(DESTDIR)$(bindir)
 endif
-	@echo "Installing keymap utilities in $(bindir):"
+	@echo "Installing keymap utilities in $(DESTDIR)$(bindir):"
 	@echo "svgakeymap:       Perl script that generates scancode conversion maps."
-	@cp utils/svgakeymap $(bindir)
+	@cp utils/svgakeymap $(DESTDIR)$(bindir)
 
 installconfig:
-	@if [ ! -d $(datadir) ]; then \
-		echo Creating configuration directory $(datadir).; \
-		mkdir $(datadir); \
-		chmod go-w $(datadir); \
-	fi
-	@if [ \( -f /usr/local/lib/libvga.config -a ! -f $(datadir)/libvga.config \) ]; then \
-		echo "Moving old config file /usr/local/lib/libvga.config to $(datadir)." ; \
-		mv -f /usr/local/lib/libvga.config $(datadir)/libvga.config; \
-	fi
-	@if [ \( -f /usr/local/lib/libvga.et4000 -a ! -f $(datadir)/libvga.et4000 \) ]; then \
-		echo "Moving old config file /usr/local/lib/libvga.et4000 to $(datadir)." ; \
-		mv -f /usr/local/lib/libvga.et4000 $(datadir)/libvga.et4000; \
-	fi
-	@if [ \( -f /usr/local/lib/libvga.ega -a ! -f $(datadir)/libvga.ega \) ]; then \
-		echo "Moving old config file /usr/local/lib/libvga.ega to $(datadir)." ; \
-		mv -f /usr/local/lib/libvga.ega $(datadir)/libvga.ega; \
-	fi
-	@if [ \( -f /etc/mach32.eeprom -a ! -f $(datadir)/mach32.eeprom \) ]; then \
-		echo Consider moving your /etc/mach32.eeprom file to $(datadir) ; \
-		echo and changing $(datadir)/libvga.config appropriately. ; \
-	fi
-	@if [ ! -f $(datadir)/libvga.config ]; then \
-		echo Installing default configuration file in $(datadir).; \
-		cp $(confdir)/libvga.config $(datadir)/libvga.config; \
-	fi
-	@if [ ! -f $(datadir)/libvga.et4000 ]; then \
-		echo Installing dynamically loaded ET4000 registers in $(datadir).; \
-		cp $(confdir)/et4000.regs $(datadir)/libvga.et4000; \
-	fi
-	@if [ ! -f $(datadir)/default.keymap ]; then \
-		echo Installing default keymap file in $(datadir).; \
-		cp $(confdir)/default.keymap $(datadir)/null.keymap; \
-	fi
-	@if [ ! -f $(datadir)/dvorak-us.keymap ]; then \
-		echo Installing Dvorak keymap file in $(datadir).; \
-		cp $(confdir)/dvorak-us.keymap $(datadir)/dvorak-us.keymap; \
+	@install -d -m755 $(DESTDIR)$(datadir)
+	@if [ ! -f $(DESTDIR)$(datadir)/libvga.config ]; then \
+		echo Installing default configuration file in $(DESTDIR)$(datadir).; \
+		cp $(confdir)/libvga.config $(DESTDIR)$(datadir)/libvga.config; \
+	fi
+	@if [ ! -f $(DESTDIR)$(datadir)/libvga.et4000 ]; then \
+		echo Installing dynamically loaded ET4000 registers in $(DESTDIR)$(datadir).; \
+		cp $(confdir)/et4000.regs $(DESTDIR)$(datadir)/libvga.et4000; \
+	fi
+	@if [ ! -f $(DESTDIR)$(datadir)/default.keymap ]; then \
+		echo Installing default keymap file in $(DESTDIR)$(datadir).; \
+		cp $(confdir)/default.keymap $(DESTDIR)$(datadir)/null.keymap; \
+	fi
+	@if [ ! -f $(DESTDIR)$(datadir)/dvorak-us.keymap ]; then \
+		echo Installing Dvorak keymap file in $(DESTDIR)$(datadir).; \
+		cp $(confdir)/dvorak-us.keymap $(DESTDIR)$(datadir)/dvorak-us.keymap; \
 	fi
 
 installman:
@@ -210,8 +155,8 @@
 lib3dkit-install:
 	(cd threeDKit/; $(MAKE) install)
 	 
-install: uninstall installheaders $(INSTALLSHAREDLIB) installconfig \
-	$(INSTALLSTATICLIB) $(INSTALLUTILS) $(INSTALLMAN) installmodule $(INSTALLDEV) \
+install: installheaders $(INSTALLSHAREDLIB) installconfig \
+	$(INSTALLSTATICLIB) $(INSTALLUTILS) $(INSTALLMAN) installmodule \
 	lib3dkit-install
 	@echo
 	@echo
diff -Nur svgalib-1.9.14.orig/doc/Makefile svgalib-1.9.14/doc/Makefile
--- svgalib-1.9.14.orig/doc/Makefile	Fri Jul  7 14:43:58 2000
+++ svgalib-1.9.14/doc/Makefile	Mon Apr 29 20:16:56 2002
@@ -65,26 +65,15 @@
 	cat $(VGAGL) | grep -v '^\.so ' | troff -man -T$@ > vgagl.$@
 	cat $(THREED) | grep -v '^\.so ' | troff -man -T$@ > 3d.$@
 
-install: gzip
-	mkdir -p $(mandir)/man1 $(mandir)/man3 $(mandir)/man5 \
-		 $(mandir)/man6 $(mandir)/man7 $(mandir)/man8
-	$(INSTALL_DATA) $(srcdir)/doc/man1/* $(mandir)/man1
-	$(INSTALL_DATA) $(srcdir)/doc/man3/* $(mandir)/man3
-	$(INSTALL_DATA) $(srcdir)/doc/man5/* $(mandir)/man5
-	$(INSTALL_DATA) $(srcdir)/doc/man6/* $(mandir)/man6
-	$(INSTALL_DATA) $(srcdir)/doc/man7/* $(mandir)/man7
-	$(INSTALL_DATA) $(srcdir)/doc/man8/* $(mandir)/man8
-	for dir in man? ; do \
-		echo gunzipping manpages in $$dir;\
-		for file in $$dir/*; do \
-			gunzip $$file; \
-		done; \
-	done
-ifdef MAKEWHATIS
-	@echo "Rebuilding the whatis database (this is slow!)"
-	@echo "You can comment this out in Makefile.cfg!"
-	$(MAKEWHATIS)
-endif
+install:
+	install -d $(DESTDIR)$(mandir)/man1 $(DESTDIR)$(mandir)/man3 $(DESTDIR)$(mandir)/man5\
+		 $(DESTDIR)$(mandir)/man6 $(DESTDIR)$(mandir)/man7 $(DESTDIR)$(mandir)/man8
+	$(INSTALL_DATA) $(srcdir)/doc/man1/* $(DESTDIR)$(mandir)/man1
+	$(INSTALL_DATA) $(srcdir)/doc/man3/* $(DESTDIR)$(mandir)/man3
+	$(INSTALL_DATA) $(srcdir)/doc/man5/* $(DESTDIR)$(mandir)/man5
+	$(INSTALL_DATA) $(srcdir)/doc/man6/* $(DESTDIR)$(mandir)/man6
+	$(INSTALL_DATA) $(srcdir)/doc/man7/* $(DESTDIR)$(mandir)/man7
+	$(INSTALL_DATA) $(srcdir)/doc/man8/* $(DESTDIR)$(mandir)/man8
 
 uninstall:
 	@echo Removing all svgalib man pages.
diff -Nur svgalib-1.9.14.orig/lrmi-0.6m/Makefile svgalib-1.9.14/lrmi-0.6m/Makefile
--- svgalib-1.9.14.orig/lrmi-0.6m/Makefile	Sun Apr 16 16:05:09 2000
+++ svgalib-1.9.14/lrmi-0.6m/Makefile	Mon Apr 29 20:16:56 2002
@@ -25,8 +25,8 @@
 	$(CC) $(CFLAGS) -o $@ $^
 
 install: mode3 vga_reset
-	install mode3 /sbin
-	install vga_reset /sbin
+	install mode3 $(DESTDIR)/sbin
+	install vga_reset $(DESTDIR)/sbin
 
 .PHONY: clean
 clean:
diff -Nur svgalib-1.9.14.orig/threeDKit/Makefile svgalib-1.9.14/threeDKit/Makefile
--- svgalib-1.9.14.orig/threeDKit/Makefile	Wed Apr  3 09:57:18 2002
+++ svgalib-1.9.14/threeDKit/Makefile	Mon Apr 29 20:16:56 2002
@@ -55,18 +55,17 @@
 	$(CC) -shared -Wl,-soname,lib3dkit.so.$(MAJOR_VER) \
 	  -o lib3dkit.so.$(VERSION) $(OBJECTS) -lm
 
-$(sharedlibdir)/lib3dkit.so.$(VERSION): lib3dkit.so.$(VERSION)
-	$(INSTALL_PROGRAM) $< $(sharedlibdir)/$<
-	(cd $(sharedlibdir); ln -sf lib3dkit.so.$(VERSION) `echo lib3dkit.so.$(VERSION) | sed 's/\.so\..*/.so/'` )
-	ldconfig
+$(DESTDIR)$(sharedlibdir)/lib3dkit.so.$(VERSION): lib3dkit.so.$(VERSION)
+	$(INSTALL_PROGRAM) $< $(DESTDIR)$(sharedlibdir)/$<
+	(cd $(DESTDIR)$(sharedlibdir); ln -sf lib3dkit.so.$(VERSION) `echo lib3dkit.so.$(VERSION) | sed 's/\.so\..*/.so/'` )
 
-install: $(sharedlibdir)/lib3dkit.so.$(VERSION) installheaders
+install: $(DESTDIR)$(sharedlibdir)/lib3dkit.so.$(VERSION) installheaders
 
 installheaders:
-	@cp 3dkit.h $(includedir)/3dkit.h
-	@chmod a+r $(includedir)/3dkit.h
-	@cp triangle.h $(includedir)/triangle.h
-	@chmod a+r $(includedir)/triangle.h
+	@cp 3dkit.h $(DESTDIR)$(includedir)/3dkit.h
+	@chmod a+r $(DESTDIR)$(includedir)/3dkit.h
+	@cp triangle.h $(DESTDIR)$(includedir)/triangle.h
+	@chmod a+r $(DESTDIR)$(includedir)/triangle.h
 
 endif	  
 
--- svgalib-1.9.18/kernel/svgalib_helper/Makefile.orig	2003-09-06 19:46:08.000000000 +0200
+++ svgalib-1.9.18/kernel/svgalib_helper/Makefile	2003-10-07 18:20:02.305378984 +0200
@@ -45,8 +45,8 @@
 svgalib_helper.o: $(svgalib_helper-objs)
 	$(LD) -r -o $@ $(svgalib_helper-objs)
 
-install: device modules_install
+install: modules_install
 
 modules_install: $(TARGET).o
-	mkdir -p /lib/modules/$(VER)/kernel/misc
-	install -m 0644 -c $(TARGET).o /lib/modules/$(VER)/kernel/misc
+	install -m 0755 -d $(DESTDIR)/lib/modules/$(VER)/kernel/misc
+	install -m 0644 -c $(TARGET).o $(DESTDIR)/lib/modules/$(VER)/kernel/misc
--- svgalib-1.9.18/kernel/svgalib_helper/Makefile.alt.orig	2003-08-03 16:42:23.000000000 +0200
+++ svgalib-1.9.18/kernel/svgalib_helper/Makefile.alt	2003-10-07 18:21:46.267574328 +0200
@@ -58,11 +58,11 @@
 $(TARGET).ko: $(TARGET).o
 	$(LD) -d -r $^ -o $@
 
-install: device modules_install
+install: modules_install
 
 modules_install: $(OBJS)
-	mkdir -p /lib/modules/$(VER)/kernel/misc 
-	install -m 0644 -c $(OBJS) /lib/modules/$(VER)/kernel/misc
+	install -m 0755 -d $(DESTDIR)/lib/modules/$(VER)/kernel/misc 
+	install -m 0644 -c $(OBJS) $(DESTDIR)/lib/modules/$(VER)/kernel/misc
 
 device:
 	rm -f /dev/svgalib_helper* /dev/svga_helper* /dev/svga /dev/svga?
