--- svgalib-1.9.20/Makefile.cfg.orig	2005-02-03 12:29:22.000000000 +0100
+++ svgalib-1.9.20/Makefile.cfg	2005-03-21 23:13:48.717311992 +0100
@@ -44,7 +44,7 @@
 # Common prefix for installation directories.
 # NOTE: This directory must exist when you start the install.
 TOPDIR=
-prefix = $(TOPDIR)/usr/local
+prefix = $(TOPDIR)/usr
 exec_prefix = $(prefix)
 
 # Directory where the shared stubs and static library will be installed.
@@ -63,7 +63,7 @@
 includedir = $(prefix)/include
 
 # Directory where the man files will be installed.
-mandir = $(prefix)/man
+mandir = $(prefix)/share/man
 
 # Target binary format.
 TARGET_FORMAT = elf
@@ -97,7 +97,7 @@
 # LIBC_MEMCPY = y
 
 # Uncomment this if you want to compile and install the static libs.
-# INSTALLSTATICLIB = installstaticlib
+INSTALLSTATICLIB = installstaticlib
 
 # Comment this out if you don't want to install the shared libs.
 # If you do not install the shared nor the static libs, 'make static'
@@ -315,10 +315,10 @@
 
 # Utilites used.
 AR	  = ar
-INSTALL_PROGRAM	= install -c -s -m 755 -o root -g bin
-INSTALL_SCRIPT	= install -c -m 755 -o root -g bin
-INSTALL_SHLIB	= install -c -m 755 -o root -g root
-INSTALL_DATA	= install -c -m 644 -o root -g root
+INSTALL_PROGRAM	= install -c -m 755
+INSTALL_SCRIPT	= install -c -m 755
+INSTALL_SHLIB	= install -c -m 755
+INSTALL_DATA	= install -c -m 644
 
 ifneq ($(ARCH),i386)
 	NO_ASM = y
