--- svgalib-1.9.23/kernel/svgalib_helper/main.c.orig	2006-01-09 11:17:11.000000000 +0100
+++ svgalib-1.9.23/kernel/svgalib_helper/main.c	2006-01-09 11:23:23.000000000 +0100
@@ -25,7 +25,6 @@
 
 #include <asm/uaccess.h>
 #include <asm/system.h>   /* cli(), *_flags */
-#include <asm/segment.h>  /* memcpy and such */
 #include <asm/io.h>
 #include <asm/pgtable.h>
 #include <asm/irq.h>
--- svgalib-1.9.23/kernel/svgalib_helper/i810.c.orig	2002-07-23 13:15:10.000000000 +0200
+++ svgalib-1.9.23/kernel/svgalib_helper/i810.c	2006-01-09 11:23:11.000000000 +0100
@@ -5,7 +5,6 @@
 #include <linux/version.h>
 #include <asm/uaccess.h>
 #include <asm/system.h>   /* cli(), *_flags */
-#include <asm/segment.h>  /* memcpy and such */
 #include <asm/io.h>
 #include <asm/pgtable.h>
 
--- svgalib-1.9.23/kernel/svgalib_helper/virtual.c.orig	2003-08-03 21:21:55.000000000 +0200
+++ svgalib-1.9.23/kernel/svgalib_helper/virtual.c	2006-01-09 11:23:30.000000000 +0100
@@ -3,7 +3,6 @@
 #include <linux/version.h>
 #include <asm/uaccess.h>
 #include <asm/system.h>   /* cli(), *_flags */
-#include <asm/segment.h>  /* memcpy and such */
 #include <asm/io.h>
 #include <asm/pgtable.h>
 #include <linux/pagemap.h>
