--- svgalib-1.9.25/kernel/svgalib_helper/main.c.old	2007-01-16 21:25:56.000000000 +0100
+++ svgalib-1.9.25/kernel/svgalib_helper/main.c	2007-01-16 21:26:34.000000000 +0100
@@ -17,7 +17,11 @@
 #include <linux/ioport.h>
 #include <linux/interrupt.h>
 #include <linux/pci.h>
+
+#ifdef CONFIG_DEVFS_FS
 #include <linux/devfs_fs_kernel.h>
+#endif
+
 #include <linux/mm.h>
 #include <linux/smp.h>
 #include <linux/smp_lock.h>
