--- svgalib-1.9.25/kernel/svgalib_helper/kernel26compat.h.orig	2008-01-23 17:27:18.941786030 +0100
+++ svgalib-1.9.25/kernel/svgalib_helper/kernel26compat.h	2008-01-23 17:28:35.145599705 +0100
@@ -83,6 +83,7 @@
 
 /* These are also not present in 2.6 kernels ... */
 #if (!defined _LINUX_DEVFS_FS_KERNEL_H) || (defined KERNEL_2_6)
+#include <linux/fs.h>
 static inline int devfs_register_chrdev (unsigned int major, const char *name,
                                          struct file_operations *fops)
 {
@@ -90,7 +91,12 @@
 }
 static inline int devfs_unregister_chrdev (unsigned int major,const char *name)
 {
-    return unregister_chrdev (major, name);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,23)    
+   return unregister_chrdev (major, name);
+#else
+    unregister_chrdev (major, name);
+    return 0;
+#endif
 }
 #endif
 
