--- svgalib-1.9.9/Makefile.cfg~	Tue Apr 24 19:54:02 2001
+++ svgalib-1.9.9/Makefile.cfg	Sat May 26 15:30:39 2001
@@ -25,7 +25,7 @@
 # Common prefix for installation directories.
 # NOTE: This directory must exist when you start the install.
 TOPDIR=
-prefix = $(TOPDIR)/usr/local
+prefix = $(TOPDIR)/usr
 exec_prefix = $(prefix)
 
 # Directory where the shared stubs and static library will be installed.
@@ -44,7 +44,7 @@
 includedir = $(prefix)/include
 
 # Directory where the man files will be installed.
-mandir = $(prefix)/man
+mandir = $(prefix)/share/man
 
 # Target binary format.
 # TARGET_FORMAT = a.out
@@ -71,7 +71,7 @@
 ROOT_VC_SHORTCUT = y
 
 # Uncomment this if you want to compile and install the static libs.
-# INSTALLSTATICLIB = installstaticlib
+INSTALLSTATICLIB = installstaticlib
 
 # Comment this out if you don't want to install the shared libs.
 # If you do not install the shared nor the static libs, 'make static'
@@ -83,7 +83,7 @@
 # a.out image is available install it in the first directory named something
 # like *aout/ in /etc/ld.so.conf.
 # If you want this, do not comment out the next line:
-INSTALLAOUTLIB = installaoutcompat
+#INSTALLAOUTLIB = installaoutcompat
 
 # Comment this out if you want to keep old shared images. Old header files,
 # library stubs and static libraries CANNOT be kept in public locations
@@ -282,6 +282,6 @@
 
 # Utilites used.
 AR	  = ar
-INSTALL_PROGRAM	= install -c -s -m 755 -o root -g bin
-INSTALL_SHLIB	= install -c -m 755 -o root -g bin
-INSTALL_DATA	= install -c -m 644 -o root -g bin
+INSTALL_PROGRAM	= install -c -m 755
+INSTALL_SHLIB	= install -c -m 755
+INSTALL_DATA	= install -c -m 644
